Index: src/core/load_save.c
===================================================================
--- src/core/load_save.c
+++ src/core/load_save.c
@@ -43,14 +43,14 @@
     SAVEPOINT(music_current_track_bank, 11),
     SAVEPOINT(music_current_track, 12),
     SAVEPOINT(music_events, 13),
     // scene
-    SAVEPOINT(current_scene, 14), SAVEPOINT(scene_stack_ptr, 15), SAVEPOINT(scene_stack, 16),
+    SAVEPOINT(current_scene, 14), SAVEPOINT(scene_stack_ptr, 15), SAVEPOINT(scene_stack, 16), SAVEPOINT(scene_stack_count, 17),
     // actors
-    SAVEPOINT(actors, 17),
-    SAVEPOINT(actors_active_head, 18), SAVEPOINT(actors_inactive_head, 19), SAVEPOINT(player_moving, 20), SAVEPOINT(player_collision_actor, 21),
+    SAVEPOINT(actors, 18),
+    SAVEPOINT(actors_active_head, 19), SAVEPOINT(actors_inactive_head, 20), SAVEPOINT(player_moving, 21), SAVEPOINT(player_collision_actor, 22),
     // system
-    SAVEPOINT(__rand_seed, 22),
+    SAVEPOINT(__rand_seed, 23),
     // terminator
     SAVEPOINTS_END
 };
 
@@ -61,9 +61,9 @@
 size_t save_blob_size;
 
 void data_init(void) BANKED {
     ENABLE_RAM_MBC5;
-    SWITCH_RAM_BANK(0, RAM_BANKS_ONLY);
+    SWITCH_RAM_BANK(1, RAM_BANKS_ONLY);
     // calculate save blob size
     save_blob_size = sizeof(save_signature) + sizeof(save_blob_size);
     for(const save_point_t * point = save_points; (point->target); point++) {
         save_blob_size += sizeof(point->size) + sizeof(point->id) + point->size;
@@ -71,12 +71,13 @@
 #ifdef BATTERYLESS
     // load from FLASH ROM
     for (UBYTE i = 0; i < SRAM_BANKS_TO_SAVE; i++) restore_sram_bank(i);
 #endif
+    SWITCH_RAM_BANK(0, RAM_BANKS_ONLY);
 }
 
 UBYTE * data_slot_address(UBYTE slot, UBYTE *bank) {
-    UWORD res = 0, res_bank = 0;
+    UWORD res = 0, res_bank = 1;
     for (UBYTE i = 0; i < slot; i++) {
         res += save_blob_size;
         if ((res + save_blob_size) > SRAM_BANK_SIZE) {
             if (++res_bank >= SRAM_BANKS_TO_SAVE) return NULL;
@@ -112,27 +113,42 @@
 #ifdef BATTERYLESS
     // save to FLASH ROM
     save_sram(SRAM_BANKS_TO_SAVE);
 #endif
+    SWITCH_RAM_BANK(0, RAM_BANKS_ONLY);
 }
 
 UBYTE data_load(UBYTE slot) BANKED {
     UBYTE data_bank, *save_data = data_slot_address(slot, &data_bank);
     if (save_data == NULL) return FALSE;
     SWITCH_RAM_BANK(data_bank, RAM_BANKS_ONLY);
-    if (SIGN_BY_PTR(save_data) != save_signature) return FALSE;
+    if (SIGN_BY_PTR(save_data) != save_signature){
+        SWITCH_RAM_BANK(0, RAM_BANKS_ONLY);
+        return FALSE;
+    }
     // seek to the first block
     save_data += sizeof(save_signature) + sizeof(save_blob_size);
     // load blocks
     for(const save_point_t * point = save_points; (point->target); point++) {
         // check chunk size
-        if (*(size_t*)save_data != point->size) return FALSE; else save_data += sizeof(point->size);
+        if (*(size_t*)save_data != point->size){
+            SWITCH_RAM_BANK(0, RAM_BANKS_ONLY);
+            return FALSE;
+        } else {
+            save_data += sizeof(point->size);
+        }
         // check chunk id
-        if (*(uint8_t*)save_data != point->id) return FALSE; else save_data += sizeof(point->id);
+        if (*(uint8_t*)save_data != point->id){
+            SWITCH_RAM_BANK(0, RAM_BANKS_ONLY);
+            return FALSE;
+        } else {
+            save_data += sizeof(point->id);
+        }
         // copy chunk data
         memcpy(point->target, save_data, point->size);
         save_data += point->size;
     }
+    SWITCH_RAM_BANK(0, RAM_BANKS_ONLY);
     // Restart music
     if (music_current_track_bank != MUSIC_STOP_BANK) {
         music_next_track = music_current_track;
     } else {
@@ -149,14 +165,19 @@
 #ifdef BATTERYLESS
     // save to FLASH ROM
     save_sram(SRAM_BANKS_TO_SAVE);
 #endif
+    SWITCH_RAM_BANK(0, RAM_BANKS_ONLY);
 }
 
 UBYTE data_peek(UBYTE slot, UINT16 idx, UWORD count, UINT16 * dest) BANKED {
     UBYTE data_bank, *save_data = data_slot_address(slot, &data_bank);
     if (save_data == NULL) return FALSE;
     SWITCH_RAM_BANK(data_bank, RAM_BANKS_ONLY);
-    if (SIGN_BY_PTR(save_data) != save_signature) return FALSE;
+    if (SIGN_BY_PTR(save_data) != save_signature){
+        SWITCH_RAM_BANK(0, RAM_BANKS_ONLY);
+        return FALSE;
+    }
     if (count) memcpy(dest, save_data + (sizeof(save_signature) + sizeof(save_blob_size) + sizeof(size_t) + sizeof(uint8_t)) + (idx << 1), count << 1);
+    SWITCH_RAM_BANK(0, RAM_BANKS_ONLY);
     return TRUE;
 }
