Index: include/vm.i
===================================================================
--- include/vm.i
+++ include/vm.i
@@ -3,12 +3,14 @@
 ;      args: big-endian
 ;      order: left-to-right (leftmost argument pushed first)
 
 ; exception ID's
-EXCEPTION_RESET         = 1
-EXCEPTION_CHANGE_SCENE  = 2
+EXCEPTION_RESET            = 1
+EXCEPTION_CHANGE_SCENE    = 2
 EXCEPTION_SAVE          = 3
 EXCEPTION_LOAD          = 4
+EXCEPTION_PUSH_SCENE_STACK = 5
+EXCEPTION_POP_SCENE_STACK = 6
 
 ; aliases
 .ARG0 = -1
 .ARG1 = -2
