Index: include/vm_exceptions.h
===================================================================
--- include/vm_exceptions.h
+++ include/vm_exceptions.h
@@ -4,7 +4,8 @@
 #define EXCEPTION_RESET 1
 #define EXCEPTION_CHANGE_SCENE 2
 #define EXCEPTION_SAVE 3
 #define EXCEPTION_LOAD 4
-#define EXCEPTION_TERMINATE 5
+#define EXCEPTION_PUSH_SCENE_STACK 5
+#define EXCEPTION_POP_SCENE_STACK 6
 
 #endif
\ No newline at end of file
