Index: src/core/vm_scene.c
===================================================================
--- src/core/vm_scene.c
+++ src/core/vm_scene.c
@@ -11,12 +11,15 @@
 
 BANKREF(VM_SCENE)
 
 void vm_scene_push(void) OLDCALL BANKED {
-    scene_stack_ptr->scene = current_scene;
-    scene_stack_ptr->pos = PLAYER.pos;
-    scene_stack_ptr->dir = PLAYER.dir;
-    scene_stack_ptr++;
+    if (scene_stack_count < SCENE_STACK_SIZE){
+        scene_stack_ptr->scene = current_scene;
+        scene_stack_ptr->pos = PLAYER.pos;
+        scene_stack_ptr->dir = PLAYER.dir;
+        scene_stack_ptr++;
+        scene_stack_count++;
+    }
 }
 
 static void raise_change_scene_exception(void) {
     vm_exception_code = EXCEPTION_CHANGE_SCENE;
@@ -27,16 +30,23 @@
     PLAYER.dir = scene_stack_ptr->dir;
 }
 
 void vm_scene_pop(void) OLDCALL BANKED {
-    scene_stack_ptr--;
-    raise_change_scene_exception();
+    if (scene_stack_count > 0){
+        scene_stack_ptr--;
+        scene_stack_count--;
+        raise_change_scene_exception();
+    }
 }
 
 void vm_scene_pop_all(void) OLDCALL BANKED {
-    scene_stack_ptr = scene_stack;
-    raise_change_scene_exception();
+    if (scene_stack_count > 0){
+        scene_stack_ptr = scene_stack;
+        scene_stack_count = 0;
+        raise_change_scene_exception();
+    }
 }
 
 void vm_scene_stack_reset(void) OLDCALL BANKED {
     scene_stack_ptr = scene_stack;
+    scene_stack_count = 0;
 }
