Index: src/core/load_save.c
===================================================================
--- src/core/load_save.c
+++ src/core/load_save.c
@@ -43,14 +43,14 @@
     SAVEPOINT(music_current_track_bank, 11),
     SAVEPOINT(music_current_track, 12),
     SAVEPOINT(music_events, 13),
     // scene
-    SAVEPOINT(current_scene, 14), SAVEPOINT(scene_stack_ptr, 15), SAVEPOINT(scene_stack, 16),
+    SAVEPOINT(current_scene, 14), SAVEPOINT(scene_stack_ptr, 15), SAVEPOINT(scene_stack, 16), SAVEPOINT(scene_stack_count, 17),
     // actors
-    SAVEPOINT(actors, 17),
-    SAVEPOINT(actors_active_head, 18), SAVEPOINT(actors_inactive_head, 19), SAVEPOINT(player_moving, 20), SAVEPOINT(player_collision_actor, 21),
+    SAVEPOINT(actors, 18),
+    SAVEPOINT(actors_active_head, 19), SAVEPOINT(actors_inactive_head, 20), SAVEPOINT(player_moving, 21), SAVEPOINT(player_collision_actor, 22),
     // system
-    SAVEPOINT(__rand_seed, 22),
+    SAVEPOINT(__rand_seed, 23),
     // terminator
     SAVEPOINTS_END
 };
 
@@ -61,9 +61,9 @@
 size_t save_blob_size;
 
 void data_init(void) BANKED {
     ENABLE_RAM_MBC5;
-    SWITCH_RAM_BANK(1, RAM_BANKS_ONLY);
+    SWITCH_RAM_BANK(2, RAM_BANKS_ONLY);
     // calculate save blob size
     save_blob_size = sizeof(save_signature) + sizeof(save_blob_size);
     for(const save_point_t * point = save_points; (point->target); point++) {
         save_blob_size += sizeof(point->size) + sizeof(point->id) + point->size;
@@ -75,9 +75,9 @@
     SWITCH_RAM_BANK(0, RAM_BANKS_ONLY);
 }
 
 UBYTE * data_slot_address(UBYTE slot, UBYTE *bank) {
-    UWORD res = 0, res_bank = 1;
+    UWORD res = 0, res_bank = 2;
     for (UBYTE i = 0; i < slot; i++) {
         res += save_blob_size;
         if ((res + save_blob_size) > SRAM_BANK_SIZE) {
             if (++res_bank >= SRAM_BANKS_TO_SAVE) return NULL;
