Index: include/data_manager.h
===================================================================
--- include/data_manager.h
+++ include/data_manager.h
@@ -25,8 +25,9 @@
 
 #define SCENE_STACK_SIZE    8
 extern scene_stack_item_t scene_stack[SCENE_STACK_SIZE];
 extern scene_stack_item_t * scene_stack_ptr;
+extern UBYTE scene_stack_count;
 
 #define MAX_SCENE_SPRITES   64
 extern UBYTE scene_sprites_base_tiles[MAX_SCENE_SPRITES];
 
