Index: include/projectiles.h
===================================================================
--- include/projectiles.h
+++ include/projectiles.h
@@ -9,10 +9,12 @@
 
 #define MAX_PROJECTILES 5
 #define MAX_PROJECTILE_DEFS 5
 
-extern projectile_def_t projectile_defs[MAX_PROJECTILES];
+extern projectile_t projectiles[MAX_PROJECTILES];
+extern projectile_def_t projectile_defs[MAX_PROJECTILE_DEFS];
 extern projectile_t *projectiles_active_head;
+extern projectile_t *projectiles_inactive_head;
 
 void projectiles_init(void) BANKED;
 void projectiles_update(void) NONBANKED;
 void projectiles_render(void) NONBANKED;
